python/pynessie/_endpoints.py

```diff
@@ -170,24 +170,15 @@ def list_tables(
     return cast(list, _get(base_url + "/trees/tree/{}/entries".format(ref), ssl_verify=ssl_verify, params=params))
 
 
-def list_logs(
-    base_url: str, ref: str, max_result_hint: Optional[int] = None, page_token: Optional[str] = None, ssl_verify: bool = True
-) -> dict:
+def list_logs(base_url: str, ssl_verify: bool = True, **filtering_args: str) -> dict:
     """Fetch a list of all logs from a known starting reference.
 
     :param base_url: base Nessie url
-    :param ref: starting reference
-    :param max_result_hint: hint for the server, maximum number of results to return
-    :param page_token: the token retrieved from a previous page returned for the same ref
     :param ssl_verify: ignore ssl errors if False
+    :param filtering_args: All of the args used to filter the log
     :return: json dict of Nessie logs
     """
-    params = {}
-    if max_result_hint:
-        params["max"] = str(max_result_hint)
-    if page_token:
-        params["pageToken"] = page_token
-    return cast(dict, _get(base_url + "/trees/tree/{}/log".format(ref), ssl_verify=ssl_verify, params=params))
+    return cast(dict, _get(base_url + "/trees/tree/{}/log".format(filtering_args.pop("ref")), ssl_verify=ssl_verify, params=filtering_args))
```

Comment:
I think we should probably leave `ref` as a named arg here. If it isn't included (ie `filtering_args.pop("ref") == None`) then we would get an error from the server. We can prevent the error on this side by ensuring its present

Reply:
sgtm